### DISCUSSION TITLE
Simplify lighting overlay selections

### DISCUSSION THREAD

**Discussion by frankie2784** [streetcomplete/StreetComplete#4640] [Feedback]
At the moment on the lighting overlay you have to click on the "Select" button and then are given the 4 lighting options to choose from. This seems to add an unnecessary step - I think it would make more sense to just show the 4 options as soon as a way is selected?![Screenshot_20221120-152308.png](https://user-images.githubusercontent.com/19356194/202885444-632fc54a-7951-4018-ae6a-c51094a787d0.png)

**Comment by mnalis** [comment]
- Duplicate of https://github.com/streetcomplete/StreetComplete/issues/4622 ?

**Reply by rhhsm** [reply]
#4622 was quickly closed because apparently it is the default interface for overlays. However I don't see much consistency in the overlays as they are all quite different (so I don't see this as a problem). This overlay is the only one for which you have to tap to select twice to be able to give an answer.

**Reply by mnalis** [reply]
> This overlay is the only one for which you have to tap to select twice to be able to give an answer.

Hm? Can you give an example on what you mean exactly, as I don't see it? I have 3 way-oriented overlays in `v49.2`:
- Street lighting
- Sidewalks
- Street parking

All of them, when way is clicked, behave the same (as far as I can tell):
- they show current state
- they allow you to click on current state
- after clicking on current state, you have to click on new answer
- you have to click checkmark to confirm changes

The only difference is that in first step "they show current state", when the state is unknown:
- `Lit` overlay shows only text `Select:` with dropdown triangle
- other overlays show repeated graphics of `?` alongside the road. 

Perhaps `Lit` overlay might be changed to show big graphics of `?` below that `Select` text, to make it visually more similar to other overlays?

**Reply by frankie2784** [reply]
> #4622 was quickly closed because apparently it is the default interface for overlays. However I don't see much consistency in the overlays as they are all quite different (so I don't see this as a problem). This overlay is the only one for which you have to tap to select twice to be able to give an answer.

Yea I'm not sure about the particular philosophy behind this. The Addresses and Shops overlays allow you to enter data immediately after clicking the feature because they are simple features. The Street parking and Sidewalks overlays have an understandable 2nd step because you need to select each side of the road separately. The Street lighting overlay has no such requirement, so could be simplified. Maybe it simplifies the codebase(?), but from a UI perspective it's a bit silly and adds unnecessary effort to enter the data.

**Reply by frankie2784** [reply]
If the philosophy is to try and keep the "feel" consistent across the overlays then, yes, mnalis's suggestion of a blank road with ?s along its length would be the way to go. But that approach (much like it is now) would, in my opinion, be substandard from the perspective of someone trying to enter data quickly and easily using this app.

(Couldn't edit my previous comment above for some reason)

**Comment by rhhsm** [comment]
@mnalis I must admit that in the lighting overlay, so far I only selected ways that showed red (i.e. without current status) and found it "a bit silly" that after selecting the way, I had to tap "select" again. For a way that already has a current state, it's nice to show it again, but it's not strictly necessary because it's already shown on the map. Maybe for the lighting overlay the current status could be shown as bolded or highlighted, but next to the other 3 common options? Like this 
![image](https://user-images.githubusercontent.com/69100427/202922161-41058cb9-6ece-40bd-bea2-861554b57f09.png) (with checkmark and three dots below it)

**Reply by mnalis** [reply]
I **personally** would like that idea, as it is faster UI! 
But I'd use for same coloring for highlighting as we do e.g. in `recycling quest`, i.e.:
![small_Screenshot_20221121_044931_de westnordost streetcomplete mn debug](https://user-images.githubusercontent.com/156656/202962150-ab018423-6af5-4f3d-8788-0b8cd9ea0094.jpg) ![small_Screenshot_20221121_045550_de westnordost streetcomplete mn debug](https://user-images.githubusercontent.com/156656/202962358-7d0a60e9-574b-442f-8449-11c2ae9301ab.jpg)

However, it is not up to me :man_shrugging:. Powers-that-be should be convinced that perhaps  "**_sometimes_**  _usability trumps consistency_", or more precisely, that `Lit overlay` is _somewhat unique_ as not needing left/right-side selection (when intermediate clicking is required for obvious reason to select side - as in `sidewalks` and `street parking` overlays), and is thus being more similar to `Address overlay`, which (kind-of) shows you current selections and allows you to edit at same time:
![small_Screenshot_20221121_050531_de westnordost streetcomplete mn debug](https://user-images.githubusercontent.com/156656/202963406-f47f2599-8e35-4636-b7b9-f16a7f254a13.jpg) ![small_Screenshot_20221121_050556_de westnordost streetcomplete mn debug](https://user-images.githubusercontent.com/156656/202963411-09f3bfc7-dcf8-44fa-bb7e-a637ae2fa328.jpg)

**Reply by rhhsm** [reply]
I fully agree, incl. the highlighting colour.

**Reply by frankie2784** [reply]
On top of this, it would be great if SC could remember (and suggest) the previous selection, to be consistent with the bike paths and footpaths overlay.
